File: LinearSystem/src/CurveFitting.cpp
#include "CurveFitting.h"

CurveFitting::CurveFitting(int n) : Matrix(n, 2) {}

void CurveFitting::readPoints(ifstream &fin)
{
    for(int i = 0; i < rows; i++)
    {
        fin >> (*this)(i,0) >> (*this)(i,1);
    }
}

int CurveFitting::size() const
{
    return rows;
}

File: LinearSystem in Python/src/CurveFitting.py
from Matrix import Matrix


class CurveFitting(Matrix):
    def __init__(self, n):
        super().__init__(n, 2)

    def read_points(self, fin):
        for i in range(self.rows):
            tokens = fin.readline().split()
            self.a[i][0] = float(tokens[0])
            self.a[i][1] = float(tokens[1])

    def size(self):
        return self.rows
